### PR TITLE
Modular Shield Generators now play well with shuttles [NO GBP]

### PR DESCRIPTION
**Pull request by moocowswag**
## About The Pull Request
So an issue with the shield generator and shuttles is that on some maps depending on how the shuttle docks, a change of direction is possible on launch and my dumb brain made it so that the machines lose their links on change direction so I moved that to the wrench act.

I also made it so the shield generator automatically turns off when it changes z levels due to shuttle movement so it doesnt leave forcefield tiles behind, it can be turned back on immediately after. I also updated the description of the generator so this isnt a hidden function.

## Why It's Good For The Game
Less buggy system and now plays well with things like white ship and evac shuttle. Modular shield generators are not designed as something you are meant to have to or even be able to individually reach all of their parts to relink/fix unless a freak accident  or sabotage happened and existing on a shuttle is not a freak accident or sabotage.

## Changelog

:cl:
fix: Modular shield generator modules no longer lose linkage when riding a shuttle
fix: Modular shield generators now gracefully turn off when being moved by a shuttle rather than leaving their projections behind, the generator's description was updated to advertise this behavior.
/:cl:

### CODE REVIEW

**Comment by moocowswag** [comment]
Forgot to state that i did touch the generator's description to advertise that it cuts off when moving z levels via shuttle.

File: code/game/machinery/modular_shield.dm [hunk 1]
```diff
@@ -514,16 +518,21 @@
...
-	disconnect_connected_through_us()
-	if(isnull(shield_generator))
-		return
-	LAZYREMOVE(shield_generator.connected_modules, (src))
-	shield_generator.calculate_boost()
-	shield_generator = null
-	update_icon_state()
+/obj/machinery/modular_shield/module/node/wrench_act(mob/living/user, obj/item/tool)
+
+	if(default_change_direction_wrench(user, tool))
```
Comment (Ghommie) [inline code/game/machinery/modular_shield.dm:524]: You should return FALSE if the statement is false, to avoid this indentation.

Comment (moocowswag) [reply code/game/machinery/modular_shield.dm:524]: doesnt it automatically return false if it doesnt get a true return?

Comment (Ghommie) [reply code/game/machinery/modular_shield.dm:524]: I mean, it should be like this:
```
	if(!default_change_direction_wrench(user, tool))
		return FALSE
```
And everything following it as normal, but with less indentation.

Comment (moocowswag) [reply code/game/machinery/modular_shield.dm:524]: oh so you want to check for inverse then do everything else, okay.

Comment (moocowswag) [reply code/game/machinery/modular_shield.dm]: done

**Comment by moocowswag** [comment]
y checks cancelled

**Comment by moocowswag** [comment]
the cutoff apparently only works for certain types of shuttles (mainly lavaland mining shuttle) so im gonna have to experiment with what type of shuttles this pr didnt actually help with and why.

**Pull request merged**